pkg/cloudevents/codec/jsoncodec.go

```diff
@@ -87,26 +98,20 @@ func jsonEncode(ctx cloudevents.EventContextReader, data interface{}) ([]byte, e
 		return nil, err
 	}
 
-	mediaType, err := ctx.GetDataMediaType()
-	if err != nil {
-		return nil, err
-	}
-	datab, err := marshalEventData(mediaType, data)
-	if err != nil {
-		return nil, err
-	}
 	if data != nil {
-		if ctx.GetDataContentEncoding() == cloudevents.Base64 {
-			buf := make([]byte, base64.StdEncoding.EncodedLen(len(datab)))
-			base64.StdEncoding.Encode(buf, datab)
-			b["data"] = []byte(strconv.QuoteToASCII(string(buf)))
-		} else if mediaType == "" || mediaType == cloudevents.ApplicationJSON {
-			b["data"] = datab
-		} else if datab[0] != byte('"') {
-			b["data"] = []byte(strconv.QuoteToASCII(string(datab)))
+		mediaType, err := ctx.GetDataMediaType()
+		if err != nil {
+			return nil, err
+		}
+		isBase64 := ctx.GetDataContentEncoding() == cloudevents.Base64
```

Comment:
Would this chunk of code make more sense to be its own function?

Reply:
This logic is integral for the structured json encoding. I added a comment to help make this clear.